ptypy/engines/ML.py

```diff
@@ -589,10 +611,101 @@ def poly_line_coeffs(self, ob_h, pr_h):
                 B += Brenorm * self.regularizer.poly_line_coeffs(
                     ob_h.storages[name].data, s.data)
 
+        if np.isinf(B).any() or np.isnan(B).any():
+            logger.warning(
+                'Warning! inf or nan found! Trying to continue...')
+            B[np.isinf(B)] = 0.
+            B[np.isnan(B)] = 0.
+
         self.B = B
 
         return B
 
+    def poly_line_all_coeffs(self, ob_h, pr_h):
```

Comment:
Wouldn't it make sense to have a single method with an internal switch based on 'quadratic' or 'all'? It feels like there is a lot of repetition between the two...

Reply:
Yes there is a lot of code duplication, but I believe @daurer preferred it this way as there's no risk of breakage...